x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx

```diff
@@ -613,3 +622,78 @@ export const DetailsPageMappingsContent: FunctionComponent<{
     </>
   );
 };
+
+function hasSemanticTextField(fields: NormalizedFields): boolean {
+  return Object.values(fields.byId).some((field) => field.source.type === 'semantic_text');
+}
+
+function getStateWithCopyToFields(state: State): State {
+  // Make sure we don't accidentally modify existing state
+  let updatedState = cloneDeep(state);
```

Comment:
Nice improvement. This indeed avoid the necessity to have a separate state variable